Source/VMware.vSphereDSC/VMware.vSphereDSC.Helper.psm1

```diff
@@ -192,4 +192,33 @@ function Update-PerfInterval
     )
 
     $PerformanceManager.UpdatePerfInterval($PerformanceInterval)
+}
+
+function Compare-Settings
+{
+    <#
+    .SYNOPSIS
+    Compare settings between current and desired states
+    .DESCRIPTION
+    This compares the current and desired states by comparing the configuration values specified in the desired state to the current state.
+    If a value is not specified in the desired state it is not assessed against the current state.
+    
+    .PARAMETER DesiredState
+    Desired state configuration object.
+    
+    .PARAMETER CurrentState
+    Current state configuration object.
+    #>
+    [CmdletBinding()]
+    param(
+        $DesiredState,
+        $CurrentState
+    )
+
+    foreach ($key in $DesiredState.Keys) {
```

Comment:
If $DesiredState is null, here NullReferenceException will be thrown.

Reply:
In testing when $DesiredState is not supplied the return has been $true. If you would like the functionality to through an exception I will add that check in.

Reply:
Yes, I saw that no exception is thrown when null is passed. At the moment leave the functionality as it is right now.